lib/scoped/index.js

```diff
@@ -5,16 +5,18 @@ var EventEmitter = require('events').EventEmitter
 function scoped (state, api, type) {
   var emitter = new EventEmitter()
 
+  var boundTypeChecker = require('../utils/type-check-wrapper').bind(null, type)
+
   var scopedApi = {
-    add: require('./add').bind(null, type, api),
-    find: require('./find').bind(null, type, api),
-    findOrAdd: require('./find-or-add').bind(null, type, api),
-    findAll: require('./find-all').bind(null, type, api),
-    update: require('./update').bind(null, type, api),
-    updateOrAdd: require('./update-or-add').bind(null, type, api),
-    updateAll: require('./update-all').bind(null, type, api),
-    remove: require('./remove').bind(null, type, api),
-    removeAll: require('./remove-all').bind(null, type, api),
+    add: boundTypeChecker(require('./add').bind(null, api)),
+    find: boundTypeChecker(require('./find').bind(null, api)),
+    findOrAdd: boundTypeChecker(require('./find-or-add').bind(null, api)),
+    findAll: require('./find-all').bind(null, api, type),
+    update: boundTypeChecker(require('./update').bind(null, api)),
+    updateOrAdd: boundTypeChecker(require('./update-or-add').bind(null, api)),
+    updateAll: boundTypeChecker(require('./update-all').bind(null, api)),
+    remove: boundTypeChecker(require('./remove').bind(null, api)),
+    removeAll: boundTypeChecker(require('./remove-all').bind(null, api)),
```

Comment:
Yes, very true.  I sometimes get a little too obsessed with the DRY principle and forget that maintainability is also important!  I'll change the methodology so that I'm just checking the type field in each method.